File: stock-market-service/src/main/java/rs/edu/raf/dto/StockMonthlyDTO.java
package rs.edu.raf.dto;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Map;

public record StockMonthlyDTO(
    @JsonProperty("Meta Data")
    Map<String, String> metaData,
    @JsonProperty("Monthly Time Series")
    Map<String, StockHistoryInfoDTO> timeSeriesMonthly
) { }

File: stock-market-service-kotlin/src/main/kotlin/org/example/stockmarketservicekotlin/dto/StockMonthlyDTO.kt
package org.example.stockmarketservicekotlin.dto

import com.fasterxml.jackson.annotation.JsonProperty

data class StockMonthlyDTO(
    @JsonProperty("Meta Data")
    val metaData: Map<String, String>,
    @JsonProperty("Monthly Time Series")
    val timeSeriesMonthly: Map<String, StockHistoryInfoDTO>
)
